src/frontend/mame/clifront.cpp

```diff
@@ -1409,37 +1439,44 @@ void cli_frontend::execute_commands(const char *exename)
-		{ CLICOMMAND_LISTCLONES,    &cli_frontend::listclones },
-		{ CLICOMMAND_LISTBROTHERS,  &cli_frontend::listbrothers },
-		{ CLICOMMAND_LISTCRC,       &cli_frontend::listcrc },
-		{ CLICOMMAND_LISTDEVICES,   &cli_frontend::listdevices },
-		{ CLICOMMAND_LISTSLOTS,     &cli_frontend::listslots },
-		{ CLICOMMAND_LISTROMS,      &cli_frontend::listroms },
-		{ CLICOMMAND_LISTSAMPLES,   &cli_frontend::listsamples },
-		{ CLICOMMAND_VERIFYROMS,    &cli_frontend::verifyroms },
-		{ CLICOMMAND_VERIFYSAMPLES, &cli_frontend::verifysamples },
-		{ CLICOMMAND_LISTMEDIA,     &cli_frontend::listmedia },
-		{ CLICOMMAND_LISTSOFTWARE,  &cli_frontend::listsoftware },
-		{ CLICOMMAND_VERIFYSOFTWARE,&cli_frontend::verifysoftware },
-		{ CLICOMMAND_ROMIDENT,      &cli_frontend::romident },
-		{ CLICOMMAND_GETSOFTLIST,   &cli_frontend::getsoftlist },
-		{ CLICOMMAND_VERIFYSOFTLIST,&cli_frontend::verifysoftlist },
+		{ CLICOMMAND_LISTXML,			0, 1, &cli_frontend::listxml },
+		{ CLICOMMAND_LISTFULL,			0, 1, &cli_frontend::listfull },
+		{ CLICOMMAND_LISTSOURCE,		0, 1, &cli_frontend::listsource },
+		{ CLICOMMAND_LISTCLONES,		0, 1, &cli_frontend::listclones },
+		{ CLICOMMAND_LISTBROTHERS,		0, 1, &cli_frontend::listbrothers },
+		{ CLICOMMAND_LISTCRC,			0, 1, &cli_frontend::listcrc },
+		{ CLICOMMAND_LISTDEVICES,		0, 1, &cli_frontend::listdevices },
+		{ CLICOMMAND_LISTSLOTS,			0, 1, &cli_frontend::listslots },
+		{ CLICOMMAND_LISTROMS,			0, 1, &cli_frontend::listroms },
+		{ CLICOMMAND_LISTSAMPLES,		0, 1, &cli_frontend::listsamples },
+		{ CLICOMMAND_VERIFYROMS,		0, 1, &cli_frontend::verifyroms },
+		{ CLICOMMAND_VERIFYSAMPLES,		0, 1, &cli_frontend::verifysamples },
+		{ CLICOMMAND_LISTMEDIA,			0, 1, &cli_frontend::listmedia },
+		{ CLICOMMAND_LISTSOFTWARE,		0, 1, &cli_frontend::listsoftware },
+		{ CLICOMMAND_VERIFYSOFTWARE,	0, 1, &cli_frontend::verifysoftware },
+		{ CLICOMMAND_ROMIDENT,			1, 1, &cli_frontend::romident },
+		{ CLICOMMAND_GETSOFTLIST,		0, 1, &cli_frontend::getsoftlist },
+		{ CLICOMMAND_VERIFYSOFTLIST,	0, 1, &cli_frontend::verifysoftlist },
 	};
 
 	// find the command
 	for (auto & info_command : info_commands)
 	{
 		if (m_options.command() == info_command.option)
 		{
-			// parse any relevant INI files before proceeding
-			const char *sysname = m_options.system_name();
-			(this->*info_command.function)((sysname[0] == 0) ? nullptr : sysname);
+			// validate argument count
+			if (m_options.command_arguments().size() < info_command.min_args)
+				throw emu_fatalerror(EMU_ERR_INVALID_CONFIG, "Auxillary verb -%s requires at least %d argument(s)", info_command.option, info_command.min_args);
+			if (m_options.command_arguments().size() > info_command.max_args)
+				throw emu_fatalerror(EMU_ERR_INVALID_CONFIG, "Auxillary verb -%s takes at most %d argument(s)", info_command.option, info_command.max_args);
+
+			// invoke the auxillary command!
+			(this->*info_command.function)(m_options.command_arguments());
```

Comment:
Is throwing an error when no argument is supplied for `-romident` really the best choice?  Wouldn't a usage hint saying it wants an archive/folder as an argument be nicer?  For example, if you forget to give `tar` a file with the `f` option, it tries to help you out:
```
% tar xf
tar: Option f requires an argument
Usage:
  List:    tar -tf <archive-filename>
  Extract: tar -xf <archive-filename>
  Create:  tar -cf <archive-filename> [filenames...]
  Help:    tar --help
```